src/mito2/src/region/opener.rs

```diff
@@ -386,7 +388,14 @@ pub(crate) async fn replay_memtable<S: LogStore>(
     let mut wal_stream = wal.scan(region_id, replay_from_entry_id, wal_options)?;
     while let Some(res) = wal_stream.next().await {
         let (entry_id, entry) = res?;
-        debug_assert!(entry_id > flushed_entry_id);
+        ensure!(
+            entry_id > flushed_entry_id,
+            StaleLogEntrySnafu {
+                region_id,
+                flushed_entry_id,
+                unexpected_entry_id: entry_id
+            }
+        );
```

Comment:
Do we need to add a mode to tolerate this and open the region? We can obsolete entries before flushed entry id in such mode.

Reply:
> Do we need to add a mode to tolerate this and open the region? We can obsolete entries before flushed entry id in such mode.

Resolved in 8e7b065f82